### PR TITLE
Create a new click object for data export

### PR DESCRIPTION
**Pull request by sophiasun0515**
We'd like to use this new object in https://github.com/CCRI-POPROX/poprox-storage/blob/main/src/poprox_storage/repositories/clicks.py so that we can write a new`get_click_within_time_window()` function to export user click data within a certain time frame.

### CODE REVIEW

File: src/poprox_concepts/domain/click_history_with_timestamp
Comment (kluver) [inline src/poprox_concepts/domain/click_history_with_timestamp:8]: A few questions/comments here:

- I think we prefer `List[...]` instead of `list[...]` (which you need to import from `typing` I think) 
- If we're using pydantic I think we would want all article types to be serializable to json and I don't know if `(UUID, datetime)` would be serializable or not. Did you test this?

Comment (sophiasun0515) [reply src/poprox_concepts/domain/click_history_with_timestamp:8]: 1. Sure, I can hold on pushing and try test this branch in the poprox-storage repo (I assume I could just uninstall poprox-concepts, and re-install it with this branch?). 

2. Thanks for catching the type! I referred to the lower-case `list()` we use in `click_history.py`, but agree typing `List` is better in that case. I'll change it.

3. Hmm yeah it's tricky if we can serialize tuple in python (I thought it should be, but after googling some looks like others transform tuple into dict first before serializing it to json). I can use `dict` type for safety!

Comment (kluver) [reply src/poprox_concepts/domain/click_history_with_timestamp:8]: An alternate approach would be to go to create a model for capturing a single click, then have a list of those click models.

Comment (karlhigley) [reply src/poprox_concepts/domain/click_history_with_timestamp:8]: I like the idea of having a `Click` model that includes the timestamp—it's usually better to create custom types with clear names than to shoehorn your data into Python primitives.

(The anti-pattern of using language primitives instead of classes/objects is sometimes called "primitive obsession," and there are a [bunch of strategies to avoid it](https://refactoring.guru/smells/primitive-obsession) depending on where it pops up.)

Comment (sophiasun0515) [reply src/poprox_concepts/domain/click_history_with_timestamp:8]: Thank you both for the input, having a simple `Click` class also sounds good to me! I'll do that.

**Comment by sophiasun0515** [comment]
> We should also probably use this new class in `ClickHistory`, so that it contains a list of `Click` objects

We can certainly do that! But the refactor will require a bunch of changes in other repos using the `ClickHistory` object, right 😄 I can identify those in `poprox-storage` and `poprox-recommender` and apply refactor accordingly. Do we also have usage in other repos?

**Comment by karlhigley** [comment]
Ah yeah, fair. Let's hold off on that then and handle it in separate PRs

**Review by karlhigley — APPROVED** [review]

**Review by kluver — APPROVED** [review]
I'm deeply torn on whether to argue that this should have an account_id. I think in most of our usages the account is known contextually (I'm thinking click histories for instance) but in a dataset I would normally see one row represented as (account, article, timestamp) and my instinct calls for that. 

Ultimately, if this suits our current needs. Go for it!

**Comment by sophiasun0515** [comment]
> I'm deeply torn on whether to argue that this should have an account_id. I think in most of our usages the account is known contextually (I'm thinking click histories for instance) but in a dataset I would normally see one row represented as (account, article, timestamp) and my instinct calls for that.
> 
> Ultimately, if this suits our current needs. Go for it!

Sure! I think we can definitely refactor again in the future (along with having the `Click` object inside `ClickHistory`, but for the data export purpose, this change should be sufficient.

After we merge this change, the next change I'll apply is on the `poprox-platform` side. Is the split already done?

**Comment by karlhigley** [comment]
> I'm deeply torn on whether to argue that this should have an account_id

It should not! Normally I'd agree that it should, but that runs counter to our requirements re: anonymization. We used to have an `account_id` in `ClickHistory,` but that was actually problematic and we removed it. At this point, the only place in `InterestProfile` that should have a user/account identifier is the `profile_id` field, and everything else needs to omit it.

**Comment by kluver** [comment]
> > I'm deeply torn on whether to argue that this should have an account_id
> 
> It should not! Normally I'd agree that it should, but that runs counter to our requirements re: anonymization. We used to have an `account_id` in `ClickHistory,` but that was actually problematic and we removed it. At this point, the only place in `InterestProfile` that should have a user/account identifier is the `profile_id` field, and everything else needs to omit it.

I think you and I might be thinking about anonymization differently. In particular, in my mind 1. the presence of an accountId field doesn't imply it must be filled, and 2. doesn't imply that we have to put an honest/accurate UUID into that field. 

Comparing that with how I think you are thinking about this -- my approach is "lets just lie!" and your approach is "lying is probably a bad design decision" and I think you're probably right here.

**Comment by karlhigley** [comment]
My approach is "let's lie as few times as possible so we can keep them straight"—the `profile_id` field is filled in with ids that are not (but are linkable back to) the `account_id`. The more places we put account/profile ids, the more places we have to redact/mask them.

**Pull request merged**
